social_media_base/models/social_account.py

```diff
@@ -0,0 +1,225 @@
+
+    # SECURITY
+    access_token = fields.Char()
+    refresh_access_token = fields.Char()
+    expire_access_token_date = fields.Date()
+    is_property_account = fields.Boolean(
+        default=False, compute="_compute_is_property_account"
+    )
+
+    @api.depends_context("uid")
+    def _compute_is_property_account(self):
+        for account in self:
+            account.is_property_account = self.env.user == account.create_uid
+
+    def update_account(self):
+        return {
+            "res_model": "wizard.social.account",
+            "views": [[False, "form"]],
+            "target": "new",
+            "type": "ir.actions.act_window",
+            "context": {
+                "default_account_id": self.id,
+                "default_media_id": self.media_id.id,
+                "social_update_account": True,
+            },
+        }
+
+    def delete_account(self):
+        """
+        Archive social media account.
+
+        This method is used to delete the social account,
+        as well as all its associated posts and campaigns.
+        It also marks all the associated posts and campaigns
+        as inactive.
+        """
+        SocialPostAccount = self.env["social.post.account"]
+        SocialPost = self.env["social.post"]
+        UtmCampaign = self.env["utm.campaign"]
+        for account in self:
+            SocialPostAccount.search([("account_id", "=", account.id)]).write(
+                {
+                    "active": False,
+                }
+            )
+            UtmCampaign.search([("account_id", "=", account.id)]).write(
+                {
+                    "active": False,
+                }
+            )
+            post_ids = SocialPost.search([("account_ids", "in", account.id)])
+            for post in post_ids:
+                if len(post.account_ids) == 1:
+                    post.write(
+                        {
+                            "active": False,
+                        }
+                    )
+            account.write(
+                {
+                    "active": False,
+                }
+            )
+
+    def _compute_display_name(self):
+        for account in self:
+            account.display_name = (
+                f"[{account.media_type.upper()}] {account.name}"
+                if account.media_type
+                else account.name
+            )
+
+    def _fields_account_url(self):
+        return []
+
+    @api.depends(lambda self: [val[0] for val in self._fields_account_url()])
+    def _compute_account_url(self):
+        for account in self:
+            for val_url in account._fields_account_url():
+                if len(val_url) < 2:
+                    continue
+                if account.media_id.media_type:
+                    account.account_url = (
+                        val_url[1] if account.media_id.media_type in val_url[0] else ""
+                    )
+                else:
+                    continue
```

Comment:
The logic here is not reasonable, but currently using break is more appropriate than continue.
```suggestion
                    break
```